handbook: Add Frontend Platform mission, vision, responsibilities (#2758)
sourcegraph/about: handbook/engineering/web/frontend-platform/index.md

    - [Felipe Janer](../../../../company/team/index.md#felipe-janer-he-him)
    - [Valery Bugakov](../../../../company/team/index.md#valery-bugakov-he-him)

## Mission

To empower all users and Sourcegraph frontend developers to achieve maximum efficiency and effectiveness, by enabling and building a first-class web experience.

## Vision

Using and developing Sourcegraph is effortless.
- We have a frictionless developer experience that empowers our frontend engineers to achieve the maximum value from the time spent on development.
- We have an intuitive, performant, and consistent user experience that makes Sourcegraph an indispensable part of every developer workflow.

## Responsibilities

1. Frontend platform:

>    - Creating and maintaining Wildcard Component Library.

Reviewer: ```suggestion
   - Creating and maintaining the Wildcard Component Library.
```

Is it possible to link to some resource about Wildcard COmponent Library? Even if it's just a Figma link for now.

Author: @AlicjaSuska Do we have anything we can link to?

Author: I'm going to go ahead and land this PR as-is, but will add a link as soon as we have something appropriate to link to :-)

>    - Owning the Sourcegraph web tech stack, tools, patterns, documentation, education.

Reviewer: ```suggestion
   - Owning the Sourcegraph web tech stack, tools, and patterns.
```

Documentation and education are covered in the point below.

   - Documentation and training material to enable product teams and new hires to quickly learn how we do web development at Sourcegraph.
   - Define and maintain how we test and deploy frontend code.
   - Ensuring an efficient and reliable frontend CI pipeline.

>    - Track, measure, and improve cross-cutting frontend metrics like bundle size, number of requests, etc.

Reviewer: ```suggestion
   - Track, measure, and improve [Web Vital](https://web.dev/vitals/) metrics.
```

Largest Contentful Paint (LCP), First Input Delay (FID) and Cumulative Layout Shift (CLS) have pretty much become the standard metrics to measure frontend performance. Should we look to leverage this? https://web.dev/vitals/

Reviewer 2: I feel like those are good to measure, but we shouldn't _restrict_ ourselves to only measuring those. They are very focused on the first load of pages, which makes sense for the majority of web pages and especially for most webpages in Google result (i.e. it makes sense that Google only cares about these). But our app has also has a lot more interactions that are not "first load", like switching between pages, switching files, etc.

Author: I've added a link to web vitals but kept the wording as it was to indicate that those aren't the only metrics we care about.

2. The core user experience of the Sourcegraph product:
    - Accessibility, navigation, and information hierarchy
    - Performance and efficiency of the core UI

>     - Syntax highlighting and code browsing
>     - Repository landing pages and file directory pages
>     - Basic code host-like features (e.g. viewing diffs, commits, history)

Reviewer: ```suggestion
    - All code browsing and code host-like views
    - Code syntax highlighting    
```

These 3 lines pretty much all relate to code browsing and code host-like views so it might just be simpler to condense them.
